### PR TITLE
ci: add benchmarking workflow

### PR DESCRIPTION
**Pull request by ningziwen**
https://github.com/runfinch/finch/issues/345

*Description of changes:*

*Testing done:*
https://runfinch.github.io/finch/dev/bench/macOS/12/arm64/
https://runfinch.github.io/finch/dev/bench/macOS/12/x86_64/
https://runfinch.github.io/finch/dev/bench/macOS/13/arm64/
https://runfinch.github.io/finch/dev/bench/macOS/13/x86_64/

- [ X ] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/benchmark.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# - When a third-party action is added (i.e., `uses`), please also add it to `download-licenses` in Makefile.
+# - When a job is added/removed/renamed, please make corresponding changes in ci-docs.yaml.
```
Comment (weikequ) [inline .github/workflows/benchmark.yaml:2]: Will this be a required workflow as part of branch protections? If so, we need to create a docs version if we're ignoring the docs and contrib folder.

File: .github/workflows/benchmark.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          if pgrep '^socket_vmnet'; then
+            sudo pkill '^socket_vmnet'
+          fi
+      - name: Install Rosetta 2
+        run: echo "A" | softwareupdate --install-rosetta || true
+      - run: brew install go lz4 automake autoconf libtool
+      - name: Build project
+        run: |
+          export PATH="/opt/homebrew/opt/libtool/libexec/gnubin:$PATH"
+          which libtool
```
Comment (weikequ) [inline .github/workflows/benchmark.yaml:55]: nit: probably don't need this line

Comment (pendo324) [reply .github/workflows/benchmark.yaml:55]: +1, I think this is copy/pasted from other actions, we can probably remove it in general

File: .github/workflows/benchmark.yaml (hunk 1 above)
Comment (ningziwen) [reply .github/workflows/benchmark.yaml:2]: I think for now workflow still needs to run the full CI.

Comment (weikequ) [reply .github/workflows/benchmark.yaml:2]: Ok, so no branch protection requirement to run this workflow, right (aka PRs don't need this workflow to pass in order to be merged)?

**Review by pendo324 — APPROVED** [review]

File: .github/workflows/benchmark.yaml (hunk 1 above)
Comment (weikequ) [reply .github/workflows/benchmark.yaml:2]: Got it. That sounds good! Thanks for confirming

**Review by weikequ — APPROVED** [review]

**Pull request merged**